### ISSUE TITLE
Map matching not working, When coordinates are inside home or office, Where missing road in OSM data.

### ISSUE THREAD

**Issue by JeetMakk** [valhalla/valhalla#4273]
My aim is to create a map-making algorithm. So I have GPS data of user travel, Valhalla map matching is perfect solution for my use case, But where GPS track route inside home, office or parking slot, Then Valhalla map matching provide the nearest road. 

Before Map Matching
![Screenshot from 2023-08-27 22-54-34](https://github.com/valhalla/valhalla/assets/43706325/c8ba6cdd-2404-4c34-a8fb-14b739cf7307)

After Map Matching
![Screenshot from 2023-08-27 22-55-46](https://github.com/valhalla/valhalla/assets/43706325/a5bb5415-fc7e-4bbd-8b3c-d2d882f83029)

So, After Map matching yellow highways tracking is **perfect** So that is not issued, But inside green area (garden) many straight lines shows, And that is not I'm assuming, Is there any solution available where Inside the green area (garden) Also shows some of the coordinates which are not noisy.

So basically need to combination of Valhalla map matching and some noisy coordinates' removal.

Any solution?

**Comment by nilsnolde** [comment]
That looks like _very_ noisy GPS data zig-zagging randomly around with dozens of meters. I'm not sure you can expect a decent map-matching result here. I'd be curious how other providers are matching this? (OSRM, HERE or so)

**Comment by JeetMakk** [comment]
> That looks like _very_ noisy GPS data zig-zagging randomly around with dozens of meters. I'm not sure you can expect a decent map-matching result here. I'd be curious how other providers are matching this? (OSRM, HERE or so)

Hello, I want to thank you first of all for your response.
Okay, so allow me to provide a sample.

Before Map Matching
![Screenshot from 2023-08-28 23-05-44](https://github.com/valhalla/valhalla/assets/43706325/6b5861d2-e6f5-4d23-b8a2-951926cbe2e7)

So this is without filter route track by user. In my use case, the user can track their full day location by (walking + bike + car). 

As a result, when I try to use map match, Here no roads or highways are found then it shows me,
![Screenshot from 2023-08-28 23-05-52](https://github.com/valhalla/valhalla/assets/43706325/f51ad70f-9aa4-45f0-ba1b-ca9eb68b6c45)

I'm not sure whether this is achievable with Valhalla, but is there a way to achieve a hybrid solution that combines map matching and a walking route? 

I'm hoping there's a better method to implement this issue

**Comment by bmapsnavigation** [comment]
Could you draw a line of what you would've expected here? I'm just wondering, because from the docs it says

`With Valhalla's Map Matching service, you can match coordinates, such as GPS locations, to roads and paths that have been mapped in OpenStreetMap. By doing this, you can turn a path into a route with narrative instructions and also get the attribute values from that matched line.`

So that would mean that all those paths in your garden probably are not mapped?
